File: 7-kyu/count-the-digit/swift/solution.swift
import Foundation

func nbDig(_ n: Int, _ d: Int) -> Int {
    let arr = (0...n).map { String($0 * $0) }.filter { $0.contains(String(d)) }
    return arr.reduce(0) { $0 + $1.reduce(0) { $0 + ($1 == Character(String(d)) ? 1 : 0) } }
}

File: 7-kyu/count-the-digit/python/solution.py
def nb_dig(n, d):
    arr = [str(i * i) if str(d) in str(i * i) else '' for i in range(n + 1)]
    return sum(1 if len(num) > 0 and ch == str(d) else 0 for num in arr for ch in num)